repondeur/zam_repondeur/models/lecture.py

```diff
@@ -255,6 +263,36 @@ def url_key(self) -> str:
             f"{self.chambre}.{self.session}.{self.texte.numero}{partie}.{self.organe}"
         )
 
+    def _split_after(self, iterable: Iterable, predicate: Callable) -> Iterable:
+        accumulator = []
+        for item in iterable:
+            accumulator.append(item)
+            if predicate(item) and accumulator:
+                yield accumulator
+                accumulator = []
+        if accumulator:
+            yield accumulator
+
+    def _next_texte_with_lecture(self, textes: Iterable["Texte"]) -> Optional["Texte"]:
+        before, *after = self._split_after(textes, lambda t: t.pk == self.texte.pk)
+        if not after:
+            return None
+
+        textes = filter(lambda t: t.lectures, after[0])
+        return next(textes) if textes else None
+
+    @property
+    def previous(self) -> Optional["Lecture"]:
+        texte = self._next_texte_with_lecture(sorted(self.dossier.textes, reverse=True))
+        # TODO: deal with PLFSS and commissions related to the same texte as seances.
+        return texte.lectures[0] if texte else None
+
+    @property
+    def next(self) -> Optional["Lecture"]:
+        texte = self._next_texte_with_lecture(self.dossier.textes)
+        # TODO: deal with PLFSS and commissions related to the same texte as seances.
+        return texte.lectures[0] if texte else None
+
```

Comment:
I find it a bit hard to understand the implementation here

Reply:
Me too 😅 

And I assume it will be even more complex with special cases like PLFSS. I will try to get a more itertools-y version